Description: fix crash when falling back from IPv6 to IPv4
 Previously, loudmouth would set socket->watch_connect to NULL when
 encountering an error. When the connection attempt (to a different IPv6)
 succeeded later on, _lm_socket_succeeded would therefore not remove the
 socket_connect_cb watcher, leading to socket_connect_cb being called upon
 socket activity and segfaulting the program.
Author: Michael Stapelberg <stapelberg@debian.org>
Bug-Debian: http://bugs.debian.org/631729
Forwarded: no
Last-Update: 2013-02-03
--- a/loudmouth/lm-socket.c
+++ b/loudmouth/lm-socket.c
@@ -529,7 +529,6 @@
 			/* error condition, but might be possible to recover
 			 * from it (by connecting to the next host) */
 			if (!_lm_socket_failed_with_error (connect_data, err)) {
-				socket->watch_connect = NULL;
 				goto out;
 			}
 		}
@@ -560,7 +559,6 @@
 				_lm_sock_close (connect_data->fd);
 				_lm_socket_failed_with_error (connect_data, err);
 
-				socket->watch_connect = NULL;
 				goto out;
 			}
 		} 
